docs/dev/pos/send_pos_orders_to_server.rst

```diff
@@ -0,0 +1,74 @@
+==============================
+ Sending POS Orders to Server
+==============================
+
+This article describes the process of sending POS Orders to odoo server and demonstrates possible usage of extending it.
+
+
+The general process is as follows:
+
+Client side:
+
+* **export_as_JSON**: combines *order data* to send to the server
+
+* then *order* is saved to browser cache
+
+* then POS tries to send data to server
+
+
+Backend side:
+
+* **create_from_ui**: data come to POS (see `example <https://github.com/odoo/odoo/blob/33f1e5f64be0113e4e3ad7cb8de373d8ab5daa7b/addons/point_of_sale/models/pos_order.py#L722-L751>`__)
+
+* **_process_order**: process order json (created records in database, etc. Example is `here <https://github.com/odoo/odoo/blob/33f1e5f64be0113e4e3ad7cb8de373d8ab5daa7b/addons/point_of_sale/models/pos_order.py#L116-L155>`__)
+
+* **_order_fields**: prepare dictionary for create method (see `example <https://github.com/odoo/odoo/blob/33f1e5f64be0113e4e3ad7cb8de373d8ab5daa7b/addons/point_of_sale/models/pos_order.py#L34-L50>`__)
+
+So, in order to pass additional information and handle it on server we need:
+
+* extend **export_as_JSON** in client side
+* extend **_process_order** in server side
+
+Let's check it on example:
+
+``Saving removed products of POS order`` module
+===============================================
+
+The module allows to add a reason on canceling order or orderline in POS.
+
+Besides,
+
+* On canceling user can click on predefined Reason and specify custom one
```

Comment:
We don't need to advertise our module here. Just say the feature related to the code we describe here